src/order/index.ts

```diff
@@ -0,0 +1,643 @@
+
+    const orderBoxName = this.getOrderBoxName(orderId, type);
+    const order = await this.getBox(orderBoxName, "Order");
+
+    if (!order) {
+      throw new Error("Order not found");
+    }
+
+    const transactions = [
+      algosdk.makeApplicationNoOpTxnFromObject({
+        sender: this.userAddress,
+        suggestedParams: sp,
+        appIndex: this.appId,
+        appArgs: [
+          encodeString(
+            type === OrderType.Limit ? "cancel_order" : "cancel_recurring_order"
+          ),
+          intToBytes(orderId)
+        ],
+        boxes: [{appIndex: 0, name: orderBoxName}],
+        foreignAssets: [Number(order.getField("asset_id"))]
+      })
+    ];
+
+    return this.setupTxnFeeAndAssignGroupId({txns: transactions, additionalFeeCount: 1});
+  }
+
+  async getPlatformFeeRate(tinyPower: number | null) {
+    try {
+      const thresholdTinyPower = await this.getGlobal(
+        TOTAL_ORDER_COUNT_KEY,
+        0,
+        this.registryAppId
+      );
+
+      if (tinyPower && tinyPower >= thresholdTinyPower) {
+        return this.getGlobal(GOVERNOR_ORDER_FEE_RATE_KEY, 0, this.registryAppId);
+      }
+
+      return this.getGlobal(ORDER_FEE_RATE_KEY, 0, this.registryAppId);
+    } catch (error: any) {
+      throw new Error(error.message);
+    }
+  }
+
+  private getOrderCount(): Promise<number> {
+    return this.appId
+      ? this.getGlobal(TOTAL_ORDER_COUNT_KEY, 0, this.appId)
+      : Promise.resolve(0);
+  }
+
+  private getOrderBoxName(id: number, type: OrderType) {
+    const orderPrefix = encodeString(type === OrderType.Limit ? "o" : "r");
+    const orderIdBytes = intToBytes(id);
+
+    return joinByteArrays(orderPrefix, orderIdBytes);
+  }
+
+  private prepareOrderAppAssetOptInTxn(
+    userAddress: string,
+    appId: number,
+    assetIds: number[],
+    suggestedParams: algosdk.SuggestedParams
+  ) {
+    return algosdk.makeApplicationNoOpTxnFromObject({
+      sender: userAddress,
+      appIndex: appId,
+      appArgs: [encodeString("asset_opt_in"), createPaddedByteArray(assetIds)],
+      suggestedParams
+    });
+  }
+
+  private prepareOrderAppAssetOptinTxnsIfNeeded(
+    assetIds: number[],
+    suggestedParams: algosdk.SuggestedParams
+  ) {
+    if (!this.appId || !this.applicationAddress) {
+      throw new Error("Application ID not provided");
+    }
+
+    if (!assetIds.length) {
+      return [];
+    }
+
+    return [
+      this.prepareOrderAppAssetOptInTxn(
+        this.userAddress,
+        this.appId,
+        assetIds,
+        suggestedParams
+      )
+    ];
+  }
+
+  async collect(orderId: number, type: OrderType) {
+    if (!this.appId) {
+      throw new Error("Application ID not provided");
+    }
+
+    const sp = await this.getSuggestedParams();
+
+    const orderBoxName = this.getOrderBoxName(orderId, type);
+    const order = await this.getBox(orderBoxName, "Order");
+
+    if (!order) {
+      throw new Error("Order not found");
+    }
+
+    const txns = [
```

Comment:
```suggestion
    const orderBoxName = this.getOrderBoxName(orderId, type);
    const order = await this.getBox(orderBoxName, "Order");

    if (!order) {
      throw new Error("Order not found");
    }

    const sp = await this.getSuggestedParams();
    const txns = [
```
This is minor but in this version in case of order is not found, we wouldn't await for suggestedParams unnecesarily